### PR TITLE
CNTRLPLANE-2511: refactor(cpo): move OAuth internal LB annotation into ReconcileService

### PR DESCRIPTION
**Pull request by bryan-cox**
## What this PR does / why we need it:

Addresses review feedback from @muraee on #8149:

1. **Moves the Azure internal LoadBalancer annotation** (`service.beta.kubernetes.io/azure-load-balancer-internal`) from `infra.go` into `oauth.ReconcileService()`, co-locating all OAuth service configuration in one place. This follows the same pattern used by `kas.ReconcileService()`.

2. **Reverts the unnecessary `switch` refactor** on the strategy type check back to the original `if serviceStrategy.Type != hyperv1.Route` form, since the switch didn't change behavior.

3. **Adds behavioral unit tests** for the new `isPrivate` parameter:
   - Public Azure LB does NOT set the internal LB annotation (negative assertion)
   - Private Azure LB sets the internal LB annotation
   - Blank LoadBalancer ingress (neither hostname nor IP) returns appropriate message

## Which issue(s) this PR fixes:

Fixes #8149 (review feedback)

## Special notes for your reviewer:

Follow-up to #8149. The `ReconcileService` function gains an `isPrivate bool` parameter so the ILB annotation logic can live alongside all other service configuration rather than being split across `infra.go` and `oauth/service.go`.

## Checklist:
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs.
- [x] This change includes unit tests.

## Summary by CodeRabbit

* **Improvements**
  * OAuth service reconciliation now correctly applies or removes Azure internal load balancer annotations based on private vs. public hosted control planes, and consistently sets hostname annotations when provided.
* **Tests**
  * Expanded OAuth service tests for Azure: private/public LB scenarios, removal of stale internal annotations, and additional service status edge cases.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/oauth/service.go [hunk 1]
```diff
@@ -64,12 +65,15 @@ func ReconcileService(svc *corev1.Service, ownerRef config.OwnerRef, strategy *h
...
+		}
 		if strategy.LoadBalancer != nil && strategy.LoadBalancer.Hostname != "" {
-			if svc.Annotations == nil {
-				svc.Annotations = map[string]string{}
-			}
 			svc.Annotations[hyperv1.ExternalDNSHostnameAnnotation] = strategy.LoadBalancer.Hostname
 		}
+		if isPrivate {
+			svc.Annotations[azureutil.InternalLoadBalancerAnnotation] = azureutil.InternalLoadBalancerValue
+		}
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply control-plane-operator/controllers/hostedcontrolplane/oauth/service.go:78]: Fixed in f00054cc0d. Added `delete(svc.Annotations, azureutil.InternalLoadBalancerAnnotation)` in the else branch and a regression test case that seeds the annotation and verifies it is removed when `isPrivate=false`.

**Comment by bryan-cox** [comment]
/verified later @xiuwang 

This can be verified with the previous PR for [CNTRLPLANE-2511](https://redhat.atlassian.net/browse/CNTRLPLANE-2511)
